tracks/rust/exercises/reverse-string/mentoring.md

```diff
@@ -0,0 +1,111 @@
+### Concepts
+
+- &str
+- String
+- Iterators
+- Iterator Methods
+
+### Reasonable solutions
+
+A reasonable solution should:
+
+- Use `.chars()`, `.rev()`, and `.collect()`
+- Be explicit with `.collect()` by using the turbofish operator `::<>`
+- Not use other iterators or iterator methods like `.into_iter()` or
+`.flat_map()`
+- Not use variables or loops
+- Not use `return`
+
+A bonus solution should:
+
+- Use the `unicode_segmentation` crate
+- Use `input.graphemes(true)` instead of
+`UnicodeSegmentation::graphemes(input, true)`
+
+### Examples
+
+```rust
+pub fn reverse(input: &str) -> String {
+    input.chars().rev().collect::<String>()
+}
+```
+
+##### Bonus solution:
+
+```rust
+use unicode_segmentation::UnicodeSegmentation;
+
+pub fn reverse(input: &str) -> String {
+    input.graphemes(true).rev().collect::<String>()
+}
+```
+
+### Example Comments
+
+If they don't use the turbofish operator:
+```
+It's best to be explicit with generics such as `.collect()` by using the turbofish operator `::<>` to specify what type we want to build. With
+more code, it makes it easier to simply look at the collect instead of having to find the function's signature.
+```
+
+If they use other iterators:
+```
+We don't need the `.into_inter()` call since `.chars()` returns an iterator.
+```
+
+If they use other iterator methods:
+```
+We don't need the `.flat_map()` call.
+```
+
+If they don't use `.rev()`:
+```
+There's a method in the standard library that reverses the direction of an iterator, in this case, produced by `.chars()`.
+```
+
+If they use variable(s):
+```
+We can `.collect()` directly into a `String` without storing the data in a variable.
+```
+
+If they use a mutatable variable and loop:
+```
+I'm glad you figured out a solution!
+
+The methodology of creating a mutatable variable and modifying it from a loop is an antiquated way of thinking, though.
+
+We can do this in a more Rust-like/idiomatic and concise manner by just using iterators and chained method calls :)
+
+Check out the [.collect()](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect) method. This will allow us to omit the loop.
+```
+
+If they use a `return`:
+```
+We don't need to use the `return` keyword here, we only use that for early returns.
+```
+
+If they haven't done the bonus:
+
+```
+Do you want to give the bonus a go? :)
+```
+
+##### For the bonus:
```

Comment:
Make this level 4 (common suggestions for the bonus).

Reply:
Done :)